src/Cake.Core/Tooling/ToolSettings.cs

```diff
@@ -16,8 +16,18 @@ public class ToolSettings
 
         /// <summary>
         /// Gets or sets the argument customization.
+        /// Argument customization is a way that lets you add, replace or reuse arguments passed to a tool.
+        /// This allows you to support new tool arguments, customize arguments or address potential argument issues.
         /// </summary>
-        /// <value>The arguments.</value>
+        /// <example>
+        /// <code>
+        /// NuGetAddSource("Cake", "https://www.myget.org/F/cake/api/v3/index.json",
+        ///     new NuGetSourcesSettings { UserName = "user", Password = "incorrect",
+        ///     ArgumentCustomization = args=&gt;args.Append("-StorePasswordInClearText")
+        /// });
+        /// </code>
+        /// </example>
+        /// <value>The arguments Func.</value>
```

Comment:
Suggestion: `The delegate used to customize the <see cref="Cake.Core.IO.ProcessArgumentBuilder" />.`.